File: solutions/0/0/0/6/p0006_s0.py
#!/usr/bin/env python3.14
# -*- coding: utf-8 -*-
""" Solution to Euler Problem 6: Sum Square Difference [Level 0]. """
from __future__ import annotations

from solver.runners import runner


@runner.main
def solve(*args: str) -> str:
    """Closed form: triangular-number square minus square-pyramidal sum, S(n)**2 - SS(n); O(1)."""
    n = runner.parse_int(args[0])

    return str((n * (n + 1) // 2) ** 2 - (2 * n + 1) * (n + 1) * n // 6)


if __name__ == "__main__":
    raise SystemExit(solve())

File: solutions/public/p0006/p0006_s0.c
/* Solution to Euler Problem 6: Sum Square Difference. */
#include "runner.h"

const char *solve(int argc, char *argv[]) {
    /* Closed form: triangular-number square minus square-pyramidal sum, S(n)^2 - SS(n); O(1). */
    static char _answer[32];
    long long n = parse_int(argv[1]);
    long long sum = n * (n + 1) / 2;
    long long square_of_sum = sum * sum;
    long long sum_of_squares = (2 * n + 1) * (n + 1) * n / 6;
    { snprintf(_answer, sizeof _answer, "%lld", (long long)(square_of_sum - sum_of_squares)); return _answer; }
}
